### PR TITLE
Fix empty whatsapp no. field

### LINKED ISSUE #2593: User Generation: Phone Number: Bug
While creating user, if the whatsapp number is not as the phone number, then while typing whatsapp numbers, the input dissappears.

The bug may be an effect of the addition of the new checkbox to mark phone number as whatsapp numbere as the same.

https://user-images.githubusercontent.com/67050716/171157798-f83360f3-1bf7-4324-8744-f1527b885828.mov

### PR DESCRIPTION
**Pull request by rabilrbl**
Close #2593

### CODE REVIEW

File: src/Components/Users/UserAdd.tsx [hunk 1]
```diff
@@ -306,7 +306,7 @@ export const UserAdd = (props: UserProps) => {
   useAbortableEffect(() => {
     if (phoneIsWhatsApp) {
       handleValueChange(state.form.phone_number, "alt_phone_number");
-    } else {
+    } else if (state.form.phone_number === state.form.alt_phone_number) {
```
Comment (gigincg) [inline src/Components/Users/UserAdd.tsx]: Is this `else` condition needed?

Isn't it better to handle it on the onChange of the checkbox?

Comment (rabilrbl) [reply src/Components/Users/UserAdd.tsx]: Agree 👍

**Comment by gigincg** [comment]
@rabilrbl I think the onChange should clear the field only when going from true-false

Otherwise there is a chance that the effect gets triggered before this and thereby clearing the field when marking it true

**Comment by gigincg** [comment]
@rabilrbl also the effect need not be triggered when alt phone changes

**Comment by nihal467** [comment]
@gigincg @rabilrbl is it good for testing, the preview is failed?

**Comment by rabilrbl** [comment]
@nihal467 waiting for @gigincg  approval

File: src/Components/Users/UserAdd.tsx [hunk 2]
```diff
@@ -553,7 +550,10 @@ export const UserAdd = (props: UserProps) => {
                 />
                 <CheckboxField
                   checked={phoneIsWhatsApp}
-                  onChange={() => setPhoneIsWhatsApp(!phoneIsWhatsApp)}
+                  onChange={(_, checked) => {
+                    setPhoneIsWhatsApp(!phoneIsWhatsApp);
```
Comment (gigincg) [inline src/Components/Users/UserAdd.tsx]: Shouldn't it be setState(checked)

**Review by gigincg — CHANGES_REQUESTED** [review]

**Comment by nihal467** [comment]
@rabilrbl @gigincg the deployment preview is failed

**Comment by nihal467** [comment]
test approved

**Pull request merged**